Release 3.2.0 (#5403)
open-formulieren/open-forms: CHANGELOG.rst

    The Dutch version of this changelog can be found :ref:`here <changelog-nl>`.

3.2.0 "Zomer" (2025-07-11)
==========================

Open Forms 3.2.0 is a feature release.

.. epigraph::

    "Zomer" is Dutch for "Summer".

This contains the changes from the alpha releases and fixes applied until the stable version.
BEFORE upgrading to 3.2.0, please read the release notes carefully and review the following
instructions.

Upgrade procedure
-----------------

# TODO-5398: which version?

> ⚠️ Before upgrading to the 3.2 release series, please ensure you upgrade to Open Forms x.y.z first.

Reviewer: I haven't squashed any migrations, so there's no minimum version requirement this time.

.. warning::

   The Camunda registration backend will be removed in Open Forms 4.0. There is no
   replacement scheduled - if you rely on this plugin, please get in touch.

.. warning::

    For the Generic JSON registration plugin, we changed the way that the data is generated.
    In case of key conflicts between static, component, and user-defined variables,
    the static variables will take precedence. Previously, the component and user-defined
    variables would override the static variables. Our validation guards against the use of
    keys that are already present in the static variables, but this does not cover old forms
    and newly-added static variables.

Major features
--------------

**🔊 Improved logging**

    We improved the logs that are emitted by the application, which enable better integration with observability
    tooling like Grafana.

>     TODO-5398: should we mention that we also prototyped a dashboard? Can people even access this?

Reviewer: Let's not - I think I want it to become a bit more mature before we do this. I think at some point we'll be able to share a good dashboard that gives you insight in your OF instance(s) and then we can link to that dashboard.

> **🔒️ Authentication using Yivi and eIDAS**

Reviewer: I think 🛂  is a better emoji :P

    We implemented support for `Yivi <https://yivi.app/>`_ and `eIDAS <https://en.wikipedia.org/wiki/EIDAS>`_

>     authentication through OIDC. With support for Yivi authentication, end-users can decide which personal

Reviewer: ```suggestion
    authentication via the OpenID Connect protocol. With support for Yivi authentication, end-users can decide which personal
```
OIDC is not a very familiar abbreviation. It should be mentioned though, because we *do* have eIDAS via SAML too.

    information they want to share with Open Forms.

    Support for eIDAS will allow European citizens without a DigiD (and/or BSN) to have access to forms which
    require authentication.

**👫 Partners component with prefill**

>     We added a partners component, where information such as initials, last name, and date of birth of a
>     partner can be shown.

Reviewer: ```suggestion
    We added a partners component, where information such as initials, last name, and date of birth of a
    partner can be shown or provided.
```

    This component can be prefilled using a new family members prefill plugin, that allows retrieving data from
    "Haal Centraal BRP personen bevragen" (version 2) or "StUF-BG" (version 3.1).

**📝 JSON schema generation**

    We added the possibility to generate a JSON schema of a form. It describes the submission data of all user-defined

>     and component variables, and can be generated for the Generic JSON and Objects API registrations as if the data
>     were sent through either of those plugins.

Reviewer: ```suggestion
    and component variables, and can be generated for the Generic JSON and Objects API registrations in the
    shape of the data produced by either of those plugins.
```

  with observability tooling like Grafana.

* [:backend:`5140`] Reworked the authentication module architecture to make it possible to add support for
  new plugins based on the OpenID Connect protocol (Yivi and eIDAS).

* [:backend:`5132`] Added support for authentication using Yivi through OIDC.

    - Allows logging in to forms using DigiD, eHerkenning, or anonymously.
    - Additional attribute groups can be defined in the Yivi configuration, and relevant ones can be selected per form.
      These groups allow end-users to, optionally, provide additional personal or company details.

# TODO-5398: add bullet points
* [:backend:`4453`] Added support for authentication using eIDAS through OIDC.

    - Amazing feature 1
    - Amazing feature 2

* [:backend:`5254`] Added new family members prefill plugin.

    - The data can be retrieved from "Haal Centraal BRP personen bevragen" (version 2) or "StUF-BG" (version 3.1).
    - Partners or children of the authenticated user can be stored in a user-defined variable.
    - The retrieved data of children can be filtered by age and whether they are deceased.

> * [:backend:`4944`, :backend:`5268`, :sdk:`824`, open-formulieren/formio-builder#225] Added partners component.

Reviewer: ```suggestion
* [:backend:`4944`, :backend:`5268`, :sdk:`824`] Added partners component.
```
builder/renderer tickets are invisible components and don't need to be linked in the changelog. Backend + SDK are our "public interfaces/components/APIs".

    - It is possible to manually add a partner, or to prefill the component using the new family members prefill plugin.

>     - Partners can be registered through the to StUF-ZDS registration.

Reviewer: ```suggestion
    - Partners can be registered through the StUF-ZDS registration.
    - Partner details are included in the email registration.
```

    - Configuration issues will be included in the digest email.

* [:backend:`4923`, :backend:`5312`, :backend:`5027`] Added JSON schema generation of a form.

    - The schema can be generated from the **Registration** tab for the Objects API and Generic JSON plugins,
      and it represents the submission data as if they were sent through either of these plugins.
    - All user-defined and component variables are included in the schema.
    - The component schemas include validation rules and a description when available.

* [:backend:`5174`] Added possibiliy to configure a description for 'zaakbetrokkenen' (registrators, cosigners, or
  partners) in the StUF-ZDS plugin.
* [:backend:`5193`] Added `exp` claim to JWT in ZGW APIs.
* [:backend:`5283`] Cleaned up the displayed columns in the admin form list to improve the UX.

**Bugfixes**

* [:backend:`5041`] Fixed components with a period in their key not being added to the data in the Generic JSON
  registration.
* Fixed hidden selectboxes component being present in the submission data as an empty object.

> * [:backend:`5326`] Fixed out-of-memory errors during email clean-up.

Author: :sunglasses:
